### PR TITLE
Fixed broken leasing for multi-arch e2e jobs

### PR DESCRIPTION
**Pull request by jaypoulz**
A change the test-platform backend unexpectedly knocked out all our jobs. The issue was that we had based our leasing model on the ovirt team, since we expected to have some complex logic built into to determine the pod configuration based on the lease. However, this proved impossible since the configuration changes we needed to make (DNS related) needed to be done in the pod itself, and thus all configuration was moved to the job templates.

Thus, we never needed the extra logic that the ovirt installer uses for our deployments. Instead, we've instead switched over to using the standard leasing model as exemplified by the old ppc64le openstack jobs.

This should be much more resilient to changes in the platform.

### CODE REVIEW

**Comment by jaypoulz** [comment]
/assign @clnperez @r4f4 @crawford
This should be ready for review. Christy - you'll need to update your 4.6 job config to conform to the new leasing. Also I'd appreciate if you fixed my broken alphabet. :) (see reorder of bastion and branch env vars)

**Comment by jaypoulz** [comment]
I lied. I snuck in one last tweak that I had forgotten to port to 4.4+.
:) Now it's ready.

**Comment by clnperez** [comment]
/lgtm

for my information @jaypoulz -- what sets the new `LEASED_RESOURCE` var? I wasn't able to find that with a github search but it sounds like there's a doc somewhere I can read.

**Comment by jaypoulz** [comment]
Update: it turns out I missed removing some locking code from the template, and that has since reveal further issues.

**Comment by jaypoulz** [comment]
@clnperez @r4f4 Review please :)

File: ci-operator/templates/openshift/installer/cluster-launch-installer-remote-libvirt-e2e.yaml [hunk 1]
```diff
@@ -59,6 +59,9 @@ parameters:
   required: false
 - name: MULTIARCH_RELEASE_IMAGE_LATEST
   required: true
+- name: USE_LEASE_CLIENT
```
Comment (clnperez) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-remote-libvirt-e2e.yaml:62]: is this used anywhere?

**Comment by clnperez** [comment]
also, saw this in a slack thread: https://ci-operator-configresolver-ui-ci.apps.ci.l2s4.p1.openshiftapps.com/help/leases#job-authors so my `LEASED_RESOURCE` question is answered. :D

File: ci-operator/templates/openshift/installer/cluster-launch-installer-remote-libvirt-e2e.yaml (hunk 1 above)
Comment (jaypoulz) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-remote-libvirt-e2e.yaml:62]: I added this is a previous run and I thought it fixed something. I'm no longer convinced it does anything, but I am loath to wait another cycle to try to remove this. Perhaps you can add it as a fixup patch to your 4.6 job? (That way we can resolve the current crisis ASAP)

**Comment by jaypoulz** [comment]
Power 4.3 & 4.4 jobs were actually successful, but there is a test that another PR needs to disable. (See https://github.com/openshift/release/pull/11505)
Power 4.5 job hit the same known image stream flakes. These will be addressed with further stability PRs.
s390x 4.3 job indicates one of the clusters on that env didn't properly teardown. I will clean that up manually.
s390x 4.4 job looks like it's going to pass
s390x 4.5 job has no chance of passing because of a known agnhost issue
s390x 4.6 job looks like it failed because the cluster actually failed to come up, which is indicative of a boot bug (and a real error, but unrelated to this PR) CC @r4f4

File: ci-operator/templates/openshift/installer/cluster-launch-installer-remote-libvirt-e2e.yaml (hunk 1 above)
Comment (jaypoulz) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-remote-libvirt-e2e.yaml:62]: Actually, I'll file a follow up PR to fix this and the bridge name bug that causes the s390x 4.3 test to fail.

Comment (clnperez) [reply ci-operator/templates/openshift/installer/cluster-launch-installer-remote-libvirt-e2e.yaml:62]: yah, i was super hesitant to even ask that question for the same reason. :D sounds good to me!

**Review by alvaroaleman — APPROVED** [review]

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**